### ISSUE TITLE
Support language aliases

### ISSUE THREAD

**Issue by resolritter** [kkoomen/vim-doge#73]
**Is your feature request related to a problem? Please describe.**

related to #72 - Ideally it would depend on the syntax of where the cursor is, not `&filetype`.

vim-doge currently [matches the generator using `&filetype`](https://github.com/kkoomen/vim-doge/blob/80d25f7ad697defde4ea1e73d059711d239229fb/autoload/doge/pattern.vim#L52). This is a problem for some languages because [they sometimes vary in how the filetype is named](https://github.com/mhartington/nvim-typescript/blob/eec19746858032eddf3630a307c1f9edf6b42d53/rplugin/python3/deoplete/sources/typescript.py#L21), even though the generator they'd refer to is the same.

**Describe the solution you'd like**

You should be able to match arbitrary languages to arbitrary generators. Something like:

```vim
let g:doge_languages_to_generators = {
  \ 'typescript.tsx': 'typescript'
  ...
  \ }
```

Notice I'm using "languages" instead of "filetype" because, ideally, this would depend on the current block's syntax, not the filetype in general (as mentioned in #72).

**Describe alternatives you've considered**

There are no easy alternatives, as far as I'm aware.

**Additional context**

**Comment by kkoomen** [comment]
> Notice I'm using "languages" instead of "filetype" because, ideally, this would depend on the current block's syntax, not the filetype in general (as mentioned in #72).

This doesn't make sense, since block-based syntax is determined by filetype and not by "language". 

<hr />

I'll implement this since it's fixing a current issue at the same time. The future variable will be `g:doge_filetype_aliases` where as the `key` will be your main filetype and the `value` should be a list of 1 or more filetypes which will be aliases of the `key`. Example:

```vim
let b:doge_filetype_aliases = {
\  'javascript': ['javascript.jsx', 'javascript.tsx', 'typescript']
\}
```

If you use the above settings in the future and you open one of these filetypes: `['javascript.jsx', 'javascript.tsx', 'typescript']` then they will behave like you're opening a `javascript` filetype.

**Comment by kkoomen** [comment]
This feature has been merged and released in [v2.2.0](https://github.com/kkoomen/vim-doge/tree/v2.2.0). See the [g:doge_filetypes_aliases](https://github.com/kkoomen/vim-doge/tree/v2.2.0#gdoge_filetype_aliases) section in the README with what is supported by default.

Feel free to submit any new issues if you experience any unwanted behavior in the future. Thanks for your contribution.
